docs/blog/_posts/2020-12-18-scala3-m3.md

```diff
@@ -0,0 +1,209 @@
+
+In this article, you will find the summary of the most important changes that took place.
+
+<!--more-->
+# SBT plugin bump
+SBT plugin was bumped to 0.5.0. Because of the changes in PR [#10607](https://github.com/lampepfl/dotty/pull/10607) this release of Scala 3 will not work with earlier versions of the SBT plugin. So if you are an early adopter of Scala 3 who has some projects in it, you will need to bump your SBT plugin version to migrate to M3.
+
+# Final syntactic tweaks
+## `as` dropped from the `given` syntax
+The following syntax is obsolete:
+
+```scala
+given intOrd as Ordering[Int]:
+  ...
+given listOrd[T: Ordering] as Ordering[List[T]]:
+  ...
+
+given Ordering[Int]:
+  ...
+given [T: Ordering] as Ordering[List[T]]:
+  ...
+
+given global as ExecutionContext = ForkJoinContext()
+given Context = ctx
+```
+
+Here is how the above is expressed now:
+
+```scala
+given intOrd: Ordering[Int] with
+  ...
+given listOrd[T: Ordering]: Ordering[List[T]] with
+  ...
+
+given Ordering[Int] with
+  ...
+given [T: Ordering]: Ordering[List[T]] with
+  ...
+
+given global: ExecutionContext = ForkJoinContext()
+given Context = ctx
+```
+
+You can find an extensive discussion on the above change in its [PR #10538](https://github.com/lampepfl/dotty/pull/10538).
+
+## Drop `as` in patterns
+Since we dropped `as` from `given`s, we lost a strong reason for having `as` at all. For this reason, we dropped them also from patterns. If previously the correct binding syntax is patterns was:
+
+```scala
+case opt as Some(foo)
+```
+
+Now it got reverted to the Scala 2 style:
+
+```scala
+case opt @ Some(foo)
+```
+
+## Switch back to the old context function closure syntax
+Previously, context function closures were written as follows:
+
+```scala
+(using s: Show[String]) => s.show("foobar")
+```
+
+Now, we have reverted to the following syntax:
+```scala
+(s: Show[String]) ?=> s.show("foobar")
+```
+
+# `Matchable` trait – a new top-level type
+The motivation for this change is the fact that it is possible to access the underlying type referenced by an opaque type using pattern matching:
+
+```scala
+val imm = IArray(1,2,3) // supposedly immutable...
+imm match
+   case a: Array[Int] => a(0) = 0  // but that's shown to be lie
+```
+
+To address this change, a new trait, `Matchable`, is introduced at the top of the type hierarchy:
+
+```scala
+abstract class Any:
+  def asInstanceOf
+  def ==
+  def !=
+  def ##
+  def equals
+  def hashCode
+  def toString
+
+trait Matchable extends Any:
+  def isInstanceOf
+  def getClass
+
+class AnyVal extends Any, Matchable
+
+class Object extends Any, Matchable
+```
+
+Matchable is currently a marker trait without any methods. Over time we might migrate methods getClass and isInstanceOf to it, since these are closely related to pattern-matching.
```

Comment:
```suggestion
`Matchable` is currently a marker trait without any methods. Over time, we intend to migrate the methods `getClass` and `isInstanceOf` to it, since these are closely related to pattern-matching.
```